Tests/UIKitTests/UITableViewExtensionsTests.swift

```diff
@@ -104,6 +104,22 @@ final class UITableViewExtensionsTests: XCTestCase {
         XCTAssertFalse(tableView.isValidIndexPath(invalidIndexPath))
     }
 
+    func testSafeScrollToIndexPath() {
+        let validIndexPathTop = IndexPath(row: 0, section: 0)
+        XCTAssertNoThrow(tableView.safeScrollToRow(at: validIndexPathTop, at: .top, animated: false))
+        XCTAssertEqual(CGPoint.zero, tableView.contentOffset)
+        let validIndexPathBottom = IndexPath(row: 7, section: 1)
+        let bottomOffset = CGPoint(x: 0, y: tableView.contentSize.height - tableView.bounds.size.height)
+        tableView.safeScrollToRow(at: validIndexPathBottom, at: .bottom, animated: false)
+        #if os(tvOS)
+            XCTAssertEqual(bottomOffset.y, tableView.contentOffset.y, accuracy: 15.0)
+        #else
+            XCTAssertEqual(bottomOffset.y, tableView.contentOffset.y, accuracy: 2.0)
+        #endif
+        let invalidIndexPath = IndexPath(row: 10, section: 0)
+        XCTAssertNoThrow(tableView.safeScrollToRow(at: invalidIndexPath, at: .bottom, animated: false))
```

Comment:
Same comment, method doesn't throw.

Reply:
Thank you @setoelkahfi, I agree with @guykogus here, statements do not throw, since we are scrolling only if the index path is valid, we can set the offset to some point and check if the scroll offset did or didn't change based on the index path, something like:

```swift
func testSafeScrollToIndexPath() {
    let validIndexPathTop = IndexPath(row: 0, section: 0)

    tableView.contentOffset = .init(x: 0, y: 100)
    XCTAssertNotEqual(tableView.contentOffset, .zero)

    tableView.scrollToRow(at: validIndexPathTop, at: .top, animated: false)
    XCTAssertEqual(tableView.contentOffset, .zero)

    let validIndexPathBottom = IndexPath(row: 7, section: 1)
    let bottomOffset = CGPoint(x: 0, y: tableView.contentSize.height - tableView.bounds.size.height)

    tableView.contentOffset = .init(x: 0, y: 200)
    XCTAssertNotEqual(tableView.contentOffset, bottomOffset)

    tableView.safeScrollToRow(at: validIndexPathBottom, at: .bottom, animated: false)
    #if os(tvOS)
        XCTAssertEqual(tableView.contentOffset.y, bottomOffset.y, accuracy: 15.0)
    #else
        XCTAssertEqual(tableView.contentOffset.y, bottomOffset.y, accuracy: 2.0)
    #endif

    let invalidIndexPath = IndexPath(row: 213, section: 21)
    tableView.contentOffset = .zero

    tableView.safeScrollToRow(at: invalidIndexPath, at: .bottom, animated: false)
    XCTAssertEqual(tableView.contentOffset, .zero)
}
```

Reply:
Updated, could you please check :) @omaralbeik